### PR TITLE
Milestone 3: Game logic

### PR DESCRIPTION
**Pull request by gurbuzertunga**
Here we created the game logic of the Tic Tac Toe game.

- Each Player uses 'X' or 'O' symbol.
- Each Player takes turns to fill the board.
- Win conditions are set in horizontal, vertical and diagonal ways.
- Draw condition is set if there is no winner.

### CODE REVIEW

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
### Status: Request changes :recycle:
#### Hi @gurbuzertunga and @mohapakram :clap: :clap:
#### Well done, you have done a good job here. However, you need to implement this :point_down: requirement mentioned in my comment below.

**_Fix this and submit for review._**

#### _Reviewed by Temitayo Adeleke._

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  def display_board
+    @board.each do |row|
+      puts
+      row.each do |block|
+        print block.to_s + ' '
+      end
+      puts
+    end
+    puts
```
Comment (lekegitrepo) [inline lib/board.rb]: **_No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
### Status: Request changes :recycle:
#### Hi @gurbuzertunga and @mohapakram :clap: :clap:
#### Well done, you have done a good job here. However, you need to implement this :point_down: requirement mentioned in my comment below.

**_Fix this and submit for review._**

#### _Reviewed by Temitayo Adeleke._

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
Hello Team :wave: 
Status: Changes Required :recycle: 

Great job on doing the last requirement! :+1: 
Thanks for your time and efforts.

Let's cover 2 more details that probably occurred after the last review :arrow_down: 

## Gameplay
:rotating_light:  When I was playing the game, I realized that at the end of the game it displays "draw" and "win" messages together.
- [x]  Please fix the end game messages. 

Here's a screenshot :camera_flash: 
drawwin

## Executable File
- [x]  At the first line in main.rb  you have :arrow_down: 
```ruby
# !/usr/bin/env ruby
```
Please delete the space between the `#` and the bang `!`

It should be :arrow_down: 
```ruby
#!/usr/bin/env ruby
```

Or you will get this error if you try to run `bin/main.rb` (without specifying **ruby**) :arrow_double_down: 

execution

### That's it team :+1: 

You can reach out to me through Slack...
With all my respect... :pray:

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## Impressive progress so far 💪! Thanks for making all the changes 👏.

### However, we need to work on some changes before we can merge.

- Right now both player can have same names. Please put a check when players are entering their names.

  Annotation 2020-08-27 115927

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Review by lekegitrepo — APPROVED** [review]
### Status: Approved :heavy_check_mark:
#### You have done a good job on this project, everything is working as required.
#### This project is approved :heavy_check_mark:

#### _Reviewed by Temitayo Adeleke._

**Pull request merged**